### PR TITLE
LPS-115274 Use openToast instead of Liferay.Notification

### PR DESCRIPTION
**Pull request by brunobasto**
This PR is a follow up on a review @wincent did a while ago: See https://github.com/wincent/liferay-portal/pull/319#discussion_r436904894

### CODE REVIEW

**Comment by brunobasto** [comment]
ci:test:sf

**Comment by brunobasto** [comment]
ci:test:relevant

**Comment by brunobasto** [comment]
ci:test:relevant

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/notifications.js [hunk 1]
```diff
@@ -12,34 +12,18 @@
...
-			duration,
-			message,
-			render: true,
-			title: Liferay.Language.get(type),
-			type,
-		});
+export function showNotification(message, type = 'success') {
+	openToast({
+		message,
+		title: Liferay.Language.get(type),
```
Comment (jbalsas) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/notifications.js:20]: I don't think this ever worked. Dynamic localization is not a thing we support.

We either need to get them from the server or to have a map with the posible type titles like:

```javascript
const TYPE_TITLE = {
    danger: Liferay.Language.get('error'),
    info: Liferay.Language.get('info'),
    success: Liferay.Language.get('success'),
}
```

Beware that titles are not always the same as type... I don't think we want to show "DANGER" in a toast :)

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/notifications.js [hunk 2]
```diff
@@ -12,34 +12,18 @@
...
+		type: 'danger',
```
Comment (jbalsas) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/notifications.js:21]: Should this be simply `type`?

**Comment by jbalsas** [comment]
Hey @brunobasto, couple of comments here... on top of that, how are these methods even meant to be used? I see:
- The `showErrorNotification` is unused
- The `showNotification` method is used 4 times in `DatasetDisplay.js`

Based on how simple this abstraction is, wondering if this is worth at all having 🤔

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/notifications.js (hunk 1 above)
Comment (brunobasto) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/notifications.js:20]: Ouch.. Didn't notice that.

Yeah, will go with the map option. Thx

**Comment by brunobasto** [comment]
Well, yeah, it's pretty simple but if you think about It, it just proved itself to be useful. We just changed the underlying toast implementation and we just had to touch one place.

**Comment by brunobasto** [comment]
See https://github.com/liferay-frontend/liferay-portal/pull/117/commits/82240ce71e6d2e2b9b7c83afc08e59e9b559de32

This `showErrorNotification` may be unnecessary. With it all we are avoiding is repeating `Liferay.Language.get('unexpected-error')`. It was already there, but I'm ok with removing it.

**Comment by jbalsas** [comment]
> we just had to touch one place.

Indeed... rather than 4 call sites... at the expense of more code shipped to the browser (transpiling is not so cheap). Not saying we need to undo this, just want to make sure we're thinking about all the consequences. We're usually quick to over-abstract, so just trying to course-correct a little, or at least make us more conscious about the tradeoffs.

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/notifications.js (hunk 2 above)
Comment (brunobasto) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/notifications.js:27]: Yeah :)

**Comment by brunobasto** [comment]
> Indeed... rather than 4 call sites... at the expense of more code shipped to the browser (transpiling is not so cheap).

Right. Yeah, that makes sense. Let's put it inline and se how it looks.

**Comment by wincent** [comment]
If you'll allow me to contribute a vacuous comment to the thoughtful discussion about trade-offs that's going on here:

Screenshot 2020-07-15 -155528-F9g38ET5@2x

Trend line looks good.

lannister-nice

**Comment by jbalsas** [comment]
Powerful stuff @wincent!

**Comment by jbalsas** [comment]
Just FYI, @brunobasto, I'm happy with this as is too, no need to inlining for now, just keeping these things in mind as we move along!

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/notifications.js [hunk 3]
```diff
@@ -12,34 +12,24 @@
...
-) {
-	if (!window.AUI) {
-		return;
-	}
+import {openToast} from 'frontend-js-web';
+
+const TYPE_TITLE = {
+	danger: Liferay.Language.get('error'),
+	info: Liferay.Language.get('info'),
+	success: Liferay.Language.get('success'),
```
Comment (jbalsas) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/notifications.js:20]: This is missing `warning`, actually :)

Comment (jbalsas) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/notifications.js:20]: https://clayui.com/docs/components/alert.html


Property | Type | Required | Default | Description
-- | -- | -- | -- | --
displayType | 'danger' \| 'info' \| 'success' \| 'warning' | false | 'info' | Determines the style of the alert.

**Comment by brunobasto** [comment]
Screen Shot 2020-07-15 at 11 04 10

How about now?

**Comment by jbalsas** [comment]
mindblown

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/notifications.js (hunk 3 above)
Comment (jbalsas) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/notifications.js]: Nevermind :)

**Comment by brunobasto** [comment]
ci:test:sf

**Comment by wincent** [comment]
> How about now?

too-good

**Comment by jbalsas** [comment]
I'm all out of gifs for now, so forwarding! :)

**Comment by jbalsas** [comment]
ci:forward

**Pull request closed without merging**